frontend/src/components/tx_status_modal.rs

```diff
@@ -0,0 +1,28 @@
+use crate::components::Button;
+use yew::prelude::*;
+use yew_icons::{Icon, IconId};
+
+#[derive(Properties, PartialEq)]
+pub struct TxStatusModalProps {
+    pub status: String,
```

Comment:
same as before, check if it is needed to carry the children into this struct. Might get away with an option<id> or another lighter type